Check for current library compatibility

It's been a while since we last checked the build against the current requirements...

First up, I see that there's a TLS error with `pip` 9.0.1 on Mac OS X. Upgrading `pip` helps, but it can't upgrade itself - you have to get it again, using e.g. `curl https://bootstrap.pypa.io/get-pip.py | python` as per https://github.com/pypa/pip/issues/5236#issuecomment-381683073

Then, there's a warning on `pytest`:

```Shell
pytest 3.5.1 has requirement py>=1.5.0, but you'll have py 1.4.34 which is incompatible.
```

so...

```Shell
pip show py
Name: py
Version: 1.4.34
Summary: library with cross-python path, ini-parsing, io, code, log facilities
Home-page: http://py.readthedocs.io/
Author: holger krekel, Ronny Pfannschmidt, Benjamin Peterson and others
Author-email: pytest-dev@python.org
License: MIT license
Location: /Users/XXXXX/XXXXX/ebu/ebu-tt-live-toolkit/venv/lib/python2.7/site-packages
Requires: 
Required-by: pytest, pytest-capturelog
$ pip install --upgrade py
Collecting py
  Downloading https://files.pythonhosted.org/packages/67/a5/f77982214dd4c8fd104b066f249adea2c49e25e8703d284382eb5e9ab35a/py-1.5.3-py2.py3-none-any.whl (84kB)
    100% |████████████████████████████████| 92kB 3.6MB/s 
Installing collected packages: py
  Found existing installation: py 1.4.34
    Uninstalling py-1.4.34:
      Successfully uninstalled py-1.4.34
Successfully installed py-1.5.3
```

Now running the tests generates:

```Shell
pytest-capturelog plugin has been merged into the core, please remove it from your requirements.
```

So remove that from `requirements.txt`.

Now we seem to be up to date, let's check the tests run... oh, we have a test fail!

```Shell
ebu_tt_live/documents/test/test_converters.py F.

...

========================================================================================== FAILURES ==========================================================================================
________________________________________________________________________ TestEBUTT3ToEBUTTDConverter.test_ericsson_1 _________________________________________________________________________

self = <test_converters.TestEBUTT3ToEBUTTDConverter testMethod=test_ericsson_1>

    def test_ericsson_1(self):
    
        xml_file = self._load_asset('converter_ericsson1.xml')
    
        self._media_clock.adjust_time(timedelta(), ebuttdt.LimitedClockTimingType('12:11:50.000').timedelta)
    
>       document = EBUTT3Document.create_from_xml(xml_file)

ebu_tt_live/documents/test/test_converters.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ebu_tt_live/documents/ebutt3.py:319: in create_from_xml
    xml_text=xml
ebu_tt_live/bindings/__init__.py:58: in CreateFromDocument
    result = raw.CreateFromDocument(*args, **kwargs)
ebu_tt_live/bindings/raw/__init__.py:85: in CreateFromDocument
    saxer.parse(io.BytesIO(xmld))
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py:110: in parse
    xmlreader.IncrementalParser.parse(self, source)
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/xmlreader.py:123: in parse
    self.feed(buffer)
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py:213: in feed
    self._parser.Parse(data, isFinal)
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py:365: in end_element_ns
    self._cont_handler.endElementNS(pair, None)
venv/lib/python2.7/site-packages/pyxb/binding/saxer.py:388: in endElementNS
    binding_object = this_state.endBindingElement()
venv/lib/python2.7/site-packages/pyxb/binding/saxer.py:226: in endBindingElement
    self.__constructElement(self.__delayedConstructor, self.__attributes, args)
venv/lib/python2.7/site-packages/pyxb/binding/saxer.py:116: in __constructElement
    self.__bindingInstance = new_object_factory(*content, **kw)
venv/lib/python2.7/site-packages/pyxb/binding/basis.py:1632: in __call__
    rv = self.typeDefinition().Factory(*args, **kw)
venv/lib/python2.7/site-packages/pyxb/binding/basis.py:306: in Factory
    rv = cls._DynamicCreate(*args, **kw)
venv/lib/python2.7/site-packages/pyxb/binding/basis.py:696: in _DynamicCreate
    return ctor(*args, **kw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pyxb.binding.datatypes.nonNegativeInteger'>, args = (), kw = {'_location': pyxb.utils.utility.Location(None, 15, 16)}, is_nil = None, from_xml = True

    def __new__ (cls, *args, **kw):
        # PyXBFactoryKeywords
        kw.pop('_validate_constraints', None)
        kw.pop('_require_value', None)
        kw.pop('_element', None)
        kw.pop('_fallback_namespace', None)
        kw.pop('_apply_attributes', None)
        is_nil = kw.pop('_nil', None)
        # ConvertArguments will remove _dom_node, _element, and
        # _apply_whitespace_facet, and it will set _from_xml.
        args = cls._ConvertArguments(args, kw)
        from_xml = kw.pop('_from_xml', False)
        if ((0 == len(args))
            and from_xml
            and not is_nil
            and issubclass(cls, _NoNullaryNonNillableNew_mixin)):
>           raise pyxb.SimpleTypeValueError(cls, args);
E           SimpleTypeValueError: Type {http://www.w3.org/2001/XMLSchema}nonNegativeInteger cannot be created from: ()

venv/lib/python2.7/site-packages/pyxb/binding/basis.py:906: SimpleTypeValueError

```

Opening this as an issue to capture all the changes needed to bring us up to date.

--- a/ebu_tt_live/config/backend.py
+++ b/ebu_tt_live/config/backend.py
@@ -137,8 +137,6 @@ class TwistedBackend(BackendBase):
 
     def _ws_create_client_factories(self, connect, producer=None, consumer=None, proxy=None):
         factory_args = {}
-        if proxy:
-            factory_args.update({'host': proxy.host, 'port': proxy.port})
         for dst in connect:
             client_factory = self._websocket.BroadcastClientFactory(
                 url=dst.geturl(),
@@ -147,6 +145,10 @@ class TwistedBackend(BackendBase):
                 **factory_args
             )
             client_factory.protocol = self._websocket.BroadcastClientProtocol
+            if proxy:
+                proxy_dict = {u'host': proxy.host, u'port': proxy.port}
+                client_factory.proxy = proxy_dict
+
             client_factory.connect()
 
     def ws_backend_producer(self, custom_producer, listen=None, connect=None, proxy=None):

--- a/ebu_tt_live/config/backend.py
+++ b/ebu_tt_live/config/backend.py
@@ -145,9 +145,7 @@ class TwistedBackend(BackendBase):
                 **factory_args
             )
             client_factory.protocol = self._websocket.BroadcastClientProtocol
-            if proxy:
-                proxy_dict = {u'host': proxy.host, u'port': proxy.port}
-                client_factory.proxy = proxy_dict
+            client_factory.proxy = proxy
 
             client_factory.connect()
 

--- a/ebu_tt_live/config/carriage.py
+++ b/ebu_tt_live/config/carriage.py
@@ -2,7 +2,6 @@ from .common import ConfigurableComponent, Namespace
 from ebu_tt_live.carriage.direct import DirectCarriageImpl
 from ebu_tt_live.carriage.websocket import WebsocketProducerCarriage, WebsocketConsumerCarriage
 from ebu_tt_live.carriage import filesystem
-from ebu_tt_live.utils import HTTPProxyConfig
 from ebu_tt_live.strings import ERR_CONF_PROXY_CONF_VALUE, ERR_NO_SUCH_COMPONENT
 from ebu_tt_live.errors import ConfigurationError
 from ebu_tt_live.strings import CFG_FILENAME_PATTERN, CFG_MESSAGE_PATTERN
@@ -134,10 +133,7 @@ def parse_proxy_address(value):
     match = proxy_regex.match(value)
     if match:
         # Ignoring the protocol part for now as it is only a http proxy
-        result = HTTPProxyConfig(
-            host=match.group('host'),
-            port=int(match.group('port'))
-        )
+        result = {u'host': match.group('host'), u'port': int(match.group('port'))}
     elif value:
         # In this case something was provided that isn't a falsy value but the parsing failed.
         raise ConfigurationError(

--- a/ebu_tt_live/utils.py
+++ b/ebu_tt_live/utils.py
@@ -358,8 +358,6 @@ class AutoRegisteringABCMeta(abc.ABCMeta):
         instance = super(AutoRegisteringABCMeta, cls).__call__(*args, **kwargs)
         return instance
 
-HTTPProxyConfig = collections.namedtuple('HTTPProxyConfig', ['host', 'port'])
-
 
 # The following section is taken from https://github.com/django/django/blob/master/django/test/utils.py
 # This is a relatively simple XML comparator implementation based on Python's minidom library.

--- a/requirements.txt
+++ b/requirements.txt
@@ -9,7 +9,6 @@ sphinx-rtd-theme
 pytest-bdd
 pytest-cov
 pytest-mock
-pytest-capturelog
 pytest-twisted
 coverage
 pytest-runner

